### PR TITLE
refactor: simpler local integration tests

### PR DESCRIPTION
**Pull request by hown3d**
**Description**

This PR tries to simplify the process of running the integration tests on a developer machine.

- remove dependency to **gsutil** and creates a package for interacting with gcs buckets.
- add **pattern matching for dockerfiles** 
  - in order to run only a portion of dockerfiles during integration testing
  - running all files can exhaust the dockerhub pull limit pretty fast
- start a **local registry** during integration-test script execution, if *LOCAL* environment variable is set. 
- dockerignore to prevent rebuilding kaniko during integration test file changes

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [x] Adds integration tests if needed.

_See [the contribution guide](../CONTRIBUTING.md) for more details._

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit tests and or integration tests added.

**Release Notes**

Simplify local integration tests

### CODE REVIEW

File: pkg/util/bucket/bucket_util.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	"io"
+	"net/url"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/GoogleContainerTools/kaniko/pkg/constants"
+	"google.golang.org/api/option"
+)
+
+// func GetBucketAndItem(context string) (string, string) {
```
Comment (imjasonh) [inline pkg/util/bucket/bucket_util.go:31]: Do we still need this method?

File: pkg/util/tar_util_test.go [hunk 2]
```diff
@@ -129,3 +130,60 @@ func createTar(testdir string, writer io.Writer) error {
...
 	return nil
 }
+
+func Test_CreateTarballOfDirectory(t *testing.T) {
+	tests := []struct {
+		name    string
+		wantF   string
+		wantErr bool
+	}{
+		{
```
Comment (imjasonh) [inline pkg/util/tar_util_test.go:140]: We don't need a table-driven test here if there's only one item. I'd just remove everything above and including `t.Run`.

File: scripts/integration-test.sh [hunk 3]
```diff
@@ -23,4 +32,31 @@ docker version
...
+  FLAGS+=("--dockerfiles-pattern=$DOCKERFILE_PATTERN")
+fi
+
+if [[ -n $LOCAL ]]; then
+  echo "running in local mode, mocking registry and gcs bucket..."
+  start_local_registry
+  #start_fake_gcs_server
+  
+  IMAGE_REPO="localhost:5000/kaniko-test"
+  # FLAGS+=(
```
Comment (imjasonh) [inline scripts/integration-test.sh:50]: Do we need these?

File: scripts/integration-test.sh [hunk 4]
```diff
@@ -23,4 +32,31 @@ docker version
...
+)
+
+if [[ -n $DOCKERFILE_PATTERN ]]; then
+  FLAGS+=("--dockerfiles-pattern=$DOCKERFILE_PATTERN")
+fi
+
+if [[ -n $LOCAL ]]; then
+  echo "running in local mode, mocking registry and gcs bucket..."
+  start_local_registry
+  #start_fake_gcs_server
```
Comment (imjasonh) [inline scripts/integration-test.sh:47]: Ditto here.

**Review by imjasonh — COMMENTED** [review]
Thanks for this contribution! Anything that makes integration tests easier to run locally will help contributors and reviewers, so thank you _so much_ for this. 🙏

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -134,33 +136,25 @@ go test ./integration -v --bucket $GCS_BUCKET --repo $IMAGE_REPO -run TestLayers
...
 
 ```shell
-make integration-test
+DOCKERFILE_PATTERN="Dockerfile_test_add*" make integration-test-run
 ```
 
-You can also run tests with `go test`, for example to run tests individually:
+This will only run dockerfiles that match the pattern `Dockerfile_test_add*`
 
-```shell
```
Comment (gabyx) [inline DEVELOPMENT.md:159]: Could we keep this line, maybe by adding it at the bottom.
Its very useful.

File: pkg/util/bucket/bucket_util.go (hunk 1 above)
Comment (hown3d) [reply pkg/util/bucket/bucket_util.go]: No, removed it.

File: pkg/util/tar_util_test.go (hunk 2 above)
Comment (hown3d) [reply pkg/util/tar_util_test.go]: Removed.

**Pull request merged**